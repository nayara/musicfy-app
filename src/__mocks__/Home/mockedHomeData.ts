export const mockedHomeData = {
  status: true,
  sections: {
    totalCount: 4,
    items: [
      {
        type: "section",
        id: "0JQ5DAroEmF9ANbLaiJ7Wx",
        title: "Recently played",
        shareUrl: "https://open.spotify.com/section/0JQ5DAroEmF9ANbLaiJ7Wx",
        contents: {
          totalCount: 1,
          items: [
            {
              type: "playlist",
              id: "37i9dQZF1DXdPec7aLTmlC",
              name: "Happy Hits!",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DXdPec7aLTmlC",
              description:
                "Hits to boost your mood and fill you with happiness!",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f00000002b55b6074da1d43715fc16d6d",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
          ],
        },
      },
      {
        type: "section",
        id: "0JQ5DAUnp4wcj0bCb3wh3S",
        title: "Made For You",
        shareUrl: "https://open.spotify.com/section/0JQ5DAUnp4wcj0bCb3wh3S",
        contents: {
          totalCount: 1,
          items: [],
        },
      },
      {
        type: "section",
        id: "0JQ5DAuChZYPe9iDhh2mJz",
        title: "Spotify Playlists",
        shareUrl: "https://open.spotify.com/section/0JQ5DAuChZYPe9iDhh2mJz",
        contents: {
          totalCount: 10,
          items: [
            {
              type: "playlist",
              id: "37i9dQZF1DXdPec7aLTmlC",
              name: "Happy Hits!",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DXdPec7aLTmlC",
              description:
                "Hits to boost your mood and fill you with happiness!",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f00000002b55b6074da1d43715fc16d6d",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DXaImRpG7HXqp",
              name: "Calming Acoustic",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DXaImRpG7HXqp",
              description: "Keep calm with instrumental acoustic tracks.",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f00000002d8fe1c03afc22a01d5109233",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DWUa8ZRTfalHk",
              name: "Pop Rising",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DWUa8ZRTfalHk",
              description: "Who's now and next in pop. Cover: Chappell Roan",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f000000024a99427709cda972f6a8d932",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DWYoYGBbGKurt",
              name: "lofi chill",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DWYoYGBbGKurt",
              description: "Lazy slow lofi beats for laid-back moments.",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f0000000263ef641f3fa47592fe798e91",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DXb9izPIc0SCS",
              name: "Indie Rock Hits",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DXb9izPIc0SCS",
              description: "True indie rock hits.",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f000000028d2c0fbb4e007f6dac9db559",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DWXe9gFZP0gtP",
              name: "Stress Relief",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DWXe9gFZP0gtP",
              description:
                "Calm your mind from anxiety with gentle piano and ambient music.",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f0000000231ca8abcbff3f5a3eb1fd339",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DX82pCGH5USnM",
              name: "Lounge - Soft House",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DX82pCGH5USnM",
              description: "Soft house music for your laidback moments.",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f0000000204c4c53344f46e15b169eedf",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DX2L0iB23Enbq",
              name: "Viral Hits",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DX2L0iB23Enbq",
              description: "Viral, trending and taking off.",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f00000002d971c6c23114fc7636dc23eb",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DWXRqgorJj26U",
              name: "Rock Classics",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DWXRqgorJj26U",
              description:
                "Rock legends & epic songs that continue to inspire generations. Cover: Foo Fighters",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f0000000278b4745cb9ce8ffe32daaf7e",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
            {
              type: "playlist",
              id: "37i9dQZF1DX4sWSpwq3LiO",
              name: "Peaceful Piano",
              shareUrl:
                "https://open.spotify.com/playlist/37i9dQZF1DX4sWSpwq3LiO",
              description:
                "Peaceful piano to help you slow down, breathe, and relax. ",
              owner: {
                type: "user",
                name: "Spotify",
              },
              images: [
                [
                  {
                    url: "https://i.scdn.co/image/ab67706f00000002d073e656e546e43bc387ad79",
                    width: null,
                    height: null,
                  },
                ],
              ],
            },
          ],
        },
      },
      {
        type: "section",
        id: "0JQ5IMCbQBLktyHz8cuH7j",
        title: "Great first audiobooks",
        shareUrl: "https://open.spotify.com/section/0JQ5IMCbQBLktyHz8cuH7j",
        contents: {
          totalCount: 42,
          items: [],
        },
      },
    ],
  },
};
