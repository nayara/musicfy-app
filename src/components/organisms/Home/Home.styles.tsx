import { styled } from "styled-components";

const Container = styled.section`
  margin: 10px;
`;

const TrackWrapper = styled.div`
  margin: 10px;
`;

export { Container, TrackWrapper };
