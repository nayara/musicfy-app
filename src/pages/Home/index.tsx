import MainLayout from "../../layouts/Main";

const HomePage = () => {
  return (
    <MainLayout>
      <h2>Home Page</h2>
    </MainLayout>
  );
};

export default HomePage;
