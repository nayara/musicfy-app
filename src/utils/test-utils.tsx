import { RenderOptions, render } from "@testing-library/react";
import { FC, ReactElement, ReactNode } from "react";
import { Providers } from "src/providers";

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <Providers>{children}</Providers>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
