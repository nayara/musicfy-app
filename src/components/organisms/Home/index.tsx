import { TSectionItem, useGetHome } from "../../../api/queries/use-get-home";
import Track from "../../../components/molecules/Track";
import * as Style from "./Home.styles";

const Home = () => {
  const { data } = useGetHome();

  if (!data) return <></>;

  return (
    <Style.Container>
      {data.sections.items.map((item: TSectionItem, index) => {
        return (
          <Style.TrackWrapper key={`section-${index}`}>
            {item.contents.items.length > 0 && (
              <Track
                id={item.id}
                title={item.title}
                contentsItems={item.contents.items}
                key={`track-${index}`}
              />
            )}
          </Style.TrackWrapper>
        );
      })}
    </Style.Container>
  );
};

export default Home;
