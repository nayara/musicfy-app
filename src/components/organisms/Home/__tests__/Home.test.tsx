import { render } from "../../../../utils/test-utils";
import Home from "..";

describe("<Home />", () => {
  it("should render home correctly", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
