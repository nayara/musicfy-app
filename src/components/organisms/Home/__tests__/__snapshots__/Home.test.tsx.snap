// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home correctly 1`] = `<div />`;
